doc/UnifiedTeamD/Unified_TeamD_HLD.md

```diff
@@ -0,0 +1,426 @@
+
+- Instead of one pipe for each LAG, a single pipe will be used in global context and used across all LAGs 
+- 2 timer fds will be created, one for timers with granularity of seconds and another for 100ms granularity. These timers will then be used to add support for existing timers in teamd using hashed wheel timer. 
+
+### Overview: What Is a Wheel Timer? 
+
+A wheel timer organizes timers in a circular buffer structure where: 
+
+- Each slot in the "wheel" represents a time interval (e.g., 100ms, 1sec). 
+- Timers are inserted into future slots based on their timeout. 
+- The wheel rotates like a clock — advancing by one slot at fixed time intervals. 
+- When the wheel lands on a slot, it fires or checks all timers in that slot. 
+
+### Key Components of a Wheel Timer
+- Tick interval – Smallest unit of time. 100ms and 1sec interval timers will be used.  
+- Number of slots – Size of the wheel, for 100ms timer 20 slots will be used and for 1sec timer 120 slots will be used. 
+- Current slot index – Rotates at each tick (like a second hand). 
+- Timer entries – Each slot holds a list of timer callbacks to fire. The existing timer callbacks will be used to handle the timer expiry. 
+
+<div align="center"> <img src=images/wheel_timers.jpg width=600 /> </div>
+
+### Teamd Config Manager
+
+Teammgrd is responsible for listening the portchannel configuration and sends IPC messages to teamd using a Unix socket.
+The format of the IPC message will be as follows:
+```
+| <opcode> | <msg_type> | <portchannel_name> | <data>
+```
+
+### Teamsyncd
+
+Teamsyncd ensures seamless synchronization of LAG status by updating both STATE_DB and APP_DB with the latest operational information. It receives link state updates from the kernel via Netlink communication, processes them, and updates the databases to ensure the system accurately reflects the current status of LAG interfaces.
+Teamsyncd uses libteam for creating netlink sockets. Abstraction apis will be implemented in libteam to store created netlink sockets accordingly. Apart from this, there will be no other change required in Teamsyncd module.
+
+### tlm_teamd
+
+tlm_teamd subscribes to the LAG table in STATE_DB to monitor changes and maintain accurate state synchronization. In the legacy design, libteamdctl was being used to fetch portchannel information from teamd and update to the STATE_DB. 
+Alternatively, now tlm_teamd communicates with teamd over a Unix socket to retrieve detailed portchannel information. It sends IPC message to teamd and gets response back which it then uses to update STATE_DB with the latest portchannel information.
```

Comment:
Thinking about this, rather than building the IPC message ourselves, I would prefer that libteamdctl be updated if necessary. Otherwise, we will be breaking a layer of abstraction that we have between swss and an external application.